### PR TITLE
Make the view options for the map view persistent by storing them in the repository

### PR DESCRIPTION
**Pull request by JamesBucanek**
These commits store the various map view options in the local repository's `userInfo` property value collection.

The map options are initialized from the repository's property values during initialization, and updated whenever the user changes them.

The window restoration code in `GIMapViewController` was removed because it is now superfluous. This also eliminates the "dual reloading" issue mentioned in `-restoreStateWithCoder:`.

I AGREE TO THE GITUP CONTRIBUTOR LICENSE AGREEMENT

### CODE REVIEW

**Comment by swisspol** [comment]
No need for `kRepositoryUserInfoKey_ViewState`, nor BOOL convenience methods: just put all the keys directly in the repo user info.

Also please follow style guide - review existing code.

**Comment by swisspol** [comment]
The first commit need to remove all the code related to the window restoration, while the 2nd one implements window settings persistency through the repo info.

**Comment by JamesBucanek** [comment]
> The first commit need to remove all the code related to the window restoration, while the 2nd one implements window settings persistency through the repo info.

I tried to do that, but I couldn't cherrypick a set of lines from the diff (which mixed up the code getting added with the code getting deleted) that would result in something that would compile. l guess I could try throwing everything away and reapplying the changes by hand...

> No need for `kRepositoryUserInfoKey_ViewState` ... just put all the keys directly in the repo user info.

I was just trying to be neat, but they could easily go into `userInfo` as individual values.

> nor BOOL convenience methods:

I have to disagree. A lot of the values have non-zero defaults and you need to test to see when the value changes; without these methods you'll end up writing the same 6 lines of code over and over again, which is poor form.

> Also please follow style guide - review existing code.

I was trying. You'll have to tell me what I missed.

**Comment by swisspol** [comment]
Just use one-liners:
```
_hidesTagTips = [[self.repository userInfoForKey:@"some_key"] boolValue];

- (void)setHidesTagTips:(BOOL)flag {
  if (flag != _hidesTagTips) {
    _hidesTagTips = flag;
    [self.repository setUserInfo:@(_hidesTagTips) forKey:@"some_key"];
    [self _reloadMap:YES];
  }
}
```

And for these settings that have a `YES` default instead of a `NO`, just negate what is written to / read from the user info.

**Comment by JamesBucanek** [comment]
OK, the second version is ready.

File: GitUpKit/Views/GIMapViewController.m [hunk 1]
```diff
@@ -25,13 +25,13 @@
 #import "GCHistory+Rewrite.h"
 #import "XLFacilityMacros.h"
 
-#define kRestorableStateKey_ShowVirtualTips @"showVirtualTips"
```
Comment (swisspol) [inline GitUpKit/Views/GIMapViewController.m]: It's best to have something like:
```
#define kPersistentViewStateKeyNamespace "GIMapViewController_"
#define kPersistentViewStateKey_ShowBranchLabels kPersistentViewStateKeyNamespace "ShowBranchLabels"
#define kPersistentViewStateKey_ShowTagTips kPersistentViewStateKeyNamespace "ShowTagTips"
```
And of course:
```
_hideTagTips = ![... kPersistentViewStateKey_ShowTagTips]
```

**Comment by JamesBucanek** [comment]
Did you want me to try an undo this last commit, or just submit a third one?

**Comment by swisspol** [comment]
You can just make a 3rd commit, merge with 2nd one and force-push.

**Comment by JamesBucanek** [comment]
Third time's the charm...

**Comment by swisspol** [comment]
Good work! Notice how the final diff is much simpler and elegant than the original one too 😄

**Comment by JamesBucanek** [comment]
It's been fun. Thanks for letting me contribute.

**Pull request merged**